### PR TITLE
Handle OOM better

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Handle OOM better in:
- `fs.readFile`
- `fs.promises.readFile`
- `fs.readFileSync`
- `Blob.prototype.text`
- `Blob.prototype.arrayBuffer`
- `Blob.prototype.json`
- `Response.prototype.text`
- `Response.prototype.arrayBuffer`
- `Response.prototype.json`
- `Request.prototype.text`
- `Request.prototype.arrayBuffer`
- `Request.prototype.json`

### How did you verify your code works?

There are some tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 04b3d2adf90a28f1149ac7b0dcab0a8cd58c4465 is building: [`#1445`](https://buildkite.com/bun/bun/builds/1445)

File: src/bun.js/bindings/bindings.zig [hunk 1]
```diff
@@ -692,6 +692,11 @@ pub const ZigString = extern struct {
     }
 
     pub fn toExternalU16(ptr: [*]const u16, len: usize, global: *JSGlobalObject) JSValue {
+        if (len > String.max_length) {
+            bun.default_allocator.free(ptr[0..len]);
```
Comment (dylan-conway) [inline src/bun.js/bindings/bindings.zig:681]: I think the caller should free this. It's not guaranteed to be allocated with `bun.default_allocator`

Comment (dylan-conway) [reply src/bun.js/bindings/bindings.zig:681]: https://github.com/oven-sh/bun/blob/df33f2b2a264912c7a742d0c03587082415fbba6/src/bun.js/webcore/blob.zig#L4915-L4919

File: src/bun.js/bindings/bindings.zig [hunk 2]
```diff
@@ -873,6 +878,11 @@ pub const ZigString = extern struct {
 
     pub fn toExternalValue(this: *const ZigString, global: *JSGlobalObject) JSValue {
         this.assertGlobal();
+        if (this.len > String.max_length) {
+            bun.default_allocator.free(@constCast(this.byteSlice()));
```
Comment (dylan-conway) [inline src/bun.js/bindings/bindings.zig:867]: same here

File: src/bun.js/bindings/bindings.zig (hunk 1 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/bindings.zig:681]: it is guaranteed. note how it never passes an allocator

File: src/bun.js/bindings/bindings.zig (hunk 2 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/bindings.zig:867]: it is guaranteed. note how it never passes an allocator

**Review by dylan-conway — APPROVED** [review]

**Review by dylan-conway — APPROVED** [review]

**Pull request merged**